app/models/group_member.rb

```diff
@@ -0,0 +1,44 @@
+# == Schema Information
+#
+# Table name: group_members
+#
+#  id         :integer          not null, primary key
+#  user_id    :integer          not null
+#  group_id   :integer          not null
+#  pending    :boolean          default(TRUE), not null
+#  created_at :datetime
+#  updated_at :datetime
+#  rank       :integer          default(0), not null
+#
+
+class GroupMember < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :group
+
+  scope :pending, -> { where(pending: true) }
+  scope :accepted, -> { where(pending: false) }
+
+  enum rank: [:pleb, :mod, :admin]
+
+  validates :user_id, uniqueness: {scope: :group_id}
+
+  after_save do
+    update_group_scores!
+    update_counter_cache!
+  end
+
+  after_destroy do
+    update_counter_cache!
+  end
+
+  def update_counter_cache!
+    self.group.update_attribute :confirmed_members_count,
```

Comment:
Atomic increment and decrement would probably be good, yeah.  I'll look into it.